### PR TITLE
[MODEXPS-204] Add possibility to run jobs without 'module enable' for tenant call

### PR DESCRIPTION
**Pull request by bekhzodkurbonboev**
## Purpose
Resolves [MODEXPS-204](https://issues.folio.org/browse/MODEXPS-204) Add possibility to run jobs without 'module enable' for tenant call

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] Check logging
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/des/config/FolioExecutionContextHelper.java [hunk 1]
```diff
@@ -53,31 +44,24 @@ public boolean isModuleRegistered() {
   }
 
   public FolioExecutionContext getFolioExecutionContext(String tenantId) {
+    Map<String, Collection<String>> tenantOkapiHeaders = new ConcurrentHashMap<>();
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/des/config/FolioExecutionContextHelper.java:47]: Why not just HashMap? I don't see usage of this variable from different threads

File: src/main/java/org/folio/des/config/ServiceConfiguration.java [hunk 2]
```diff
@@ -149,7 +149,7 @@ EdifactOrdersExportJobScheduler edifactOrdersExportJobScheduler(ScheduledTaskBui
 
   @Bean
   AcqSchedulingProperties acqSchedulingProperties(
-                    @Value("${folio.schedule.acquisition.runOnlyIfModuleRegistered:true}") String runOnlyIfModuleRegistered) {
+                    @Value("${folio.schedule.acquisition.runOnlyIfModuleRegistered:false}") String runOnlyIfModuleRegistered) {
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/des/config/ServiceConfiguration.java:152]: This is just for debugging purposes, should be removed before the commit

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ folio:
   schedule:
     acquisition:
       poolSize: 10
-      runOnlyIfModuleRegistered: true
+      runOnlyIfModuleRegistered: false
```
Comment (tatsiana-tarhonskaya) [inline src/main/resources/application.yml:10]: Same, that's just for debugging purposes. This is used for inmemory scheduling and should be left as is for the commit. Quartz impl does not use this property

File: src/main/java/org/folio/des/config/FolioExecutionContextHelper.java [hunk 4]
```diff
@@ -32,18 +31,10 @@ public class FolioExecutionContextHelper {
...
-  private final Map<String, Map<String, Collection<String>>> okapiHeaders = new ConcurrentHashMap<>();
-
-  public void storeOkapiHeaders() {
-    if (MapUtils.isNotEmpty(folioExecutionContext.getOkapiHeaders())) {
-      log.info("Got OKAPI headers.");
-      okapiHeaders.put(folioExecutionContext.getTenantId(), folioExecutionContext.getOkapiHeaders());
-    }
-  }
+  // TODO find a way to get current 'x-okapi-url', might be list of urls
+  private final String okapiUrl = "http://10.0.2.15:9130";
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/des/config/FolioExecutionContextHelper.java:35]: Let's take it from the env variable and set some default

File: src/main/java/org/folio/des/config/FolioExecutionContextHelper.java (hunk 1 above)
Comment (bekhzodkurbonboev) [reply src/main/java/org/folio/des/config/FolioExecutionContextHelper.java]: Thank you, replaced.

File: src/main/java/org/folio/des/config/ServiceConfiguration.java (hunk 2 above)
Comment (bekhzodkurbonboev) [reply src/main/java/org/folio/des/config/ServiceConfiguration.java]: Fixed.

File: src/main/resources/application.yml (hunk 3 above)
Comment (bekhzodkurbonboev) [reply src/main/resources/application.yml]: Fixed.

File: src/main/java/org/folio/des/config/FolioExecutionContextHelper.java (hunk 4 above)
Comment (bekhzodkurbonboev) [reply src/main/java/org/folio/des/config/FolioExecutionContextHelper.java]: Thank you, fixed.

File: src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    wireMockServer.stubFor(
+      get(urlEqualTo("/users?query=username%3D%3Ddata-export-system-user"))
+        .willReturn(aResponse()
+          .withHeader("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+          .withBody(SYSTEM_USER)));
+
+    // 'execution context' should be created according to 'data-export-system-user' headers
+    FolioExecutionContext executionContext = contextHelper.getFolioExecutionContext(TENANT);
+
+    Assertions.assertEquals(executionContext.getTenantId(), TENANT);
```
Comment (tatsiana-tarhonskaya) [inline src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java:68]: we usually use static imports for asserts

File: src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      "resultInfo": {
+        "totalRecords": 1,
+        "facets": [],
+        "diagnostics": []
+      }
+    }
+    """;
+
+  @Test
+  void shouldGetFolioExecutionContextHelper() {
```
Comment (tatsiana-tarhonskaya) [inline src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java:51]: maybe just shouldGetFolioExecutionContext?

File: src/main/java/org/folio/des/security/AuthService.java [hunk 7]
```diff
@@ -32,20 +35,28 @@ public SystemUserParameters loginSystemUser(String tenant, String url) {
...
       log.error("Can't get token logging in as {}.", username);
     }
-    return userParameters;
+    return userParameters.getOkapiToken();
+  }
+
+  public String getSystemUserId() {
+    Optional<User> optionalUser = usersClient.getUsersByQuery("username==" + username).getUsers().stream().findFirst();
+
+    if(optionalUser.isEmpty()) {
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/des/security/AuthService.java:55]: missed space. Please, format the lines you've added

File: src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+  @Test
+  void shouldGetExceptionWhenThereIsNoToken() {
+    // request to get response without 'XOkapiHeaders.TOKEN'
+    wireMockServer.stubFor(
+      post(urlEqualTo("/authn/login"))
+        .willReturn(aResponse()));
+
+    // should get exception because of missing token
+    Assertions.assertThrows(IllegalStateException.class, () -> contextHelper.getFolioExecutionContext(TENANT));
```
Comment (tatsiana-tarhonskaya) [inline src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java:82]: Exception message can be verified as well

**Review by tatsiana-tarhonskaya — APPROVED** [review]

**Review by SerhiiNosko — APPROVED** [review]

File: src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java (hunk 5 above)
Comment (bekhzodkurbonboev) [reply src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java]: Fixed

File: src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java (hunk 6 above)
Comment (bekhzodkurbonboev) [reply src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java]: Thank you, corrected

File: src/main/java/org/folio/des/security/AuthService.java (hunk 7 above)
Comment (bekhzodkurbonboev) [reply src/main/java/org/folio/des/security/AuthService.java]: Fixed

File: src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java (hunk 8 above)
Comment (bekhzodkurbonboev) [reply src/test/java/org/folio/des/config/FolioExecutionContextHelperTest.java]: Thank you, added verification for it.

**Pull request merged**